src/viem/saga.ts

```diff
@@ -199,36 +201,33 @@ export function* sendPayment({
  */
 function* getTransferSimulateContract({
   wallet,
-  tokenId,
+  tokenAddress,
+  tokenInfo,
```

Comment:
this raises the rather uncomfortable possibility that tokenAddress will be different from the address in tokenInfo

Reply:
Fortunately, using `TokenBalanceWithAddress` means we can safely get rid of this parameter.